idaes_examples/notebooks/docs/unit_models/custom_unit_models/creating_unit_model.ipynb

```diff
@@ -0,0 +1,2813 @@
+    "\n",
+    "To create a custom initialization routine, model developers must create an initialize method as part of their model, and provide a sequence of steps intended to build up a feasible solution. Initialization routines generally make use of Pyomo’s tools for activating and deactivating constraints and often involve solving multiple sub-problems whilst building up an initial state.\n",
+    "\n",
+    "For this tutorial we would use the pre-defined initialization routine of `BlockTriangularizationInitializer` when initializing the model in the flowsheet. This Initializer should be suitable for most models, but may struggle to initialize\n",
+    "tightly coupled systems of equations. This method of initialization will follow the following workflow. \n",
+    "\n",
+    "- Have precheck for structural singularity\n",
+    "- Run incidence analysis on given block data and check matching.\n",
+    "- Call Block Triangularization solver on model.\n",
+    "- Call solve_strongly_connected_components on a given BlockData.\n",
+    "\n",
+    "For more details about this initialization routine can be found [here](https://github.com/IDAES/idaes-pse/blob/c09433b9afed5ae2fe25c0ccdc732783324f0101/idaes/core/initialization/block_triangularization.py). \n",
+    "\n",
+    "\n",
+    "This marks the conclusion of creating a custom unit model, for a more detailed explanation on creating a unit model refer [this resource](../../unit_models/custom_unit_models/custom_compressor.ipynb). The next sections will deal with the diagonistics and testing of the property package and unit model. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9293195c",
+   "metadata": {},
+   "source": [
+    "## 3.3 Building a Flowsheet\n",
+    "\n",
+    "Once we have set up the unit model and its property packages, we can start building a flowsheet using them. In this tutorial, we're focusing on a simple flowsheet with just a liquid-liquid extractor. To create the flowsheet we follow the following steps:\n",
+    "\n",
+    "- Import necessary libraries\n",
+    "- Create a Pyomo model.\n",
+    "- Inside the model, create a flowsheet block.\n",
+    "- Assign property packages to the flowsheet block.\n",
+    " -Add the liquid-liquid extractor to the flowsheet block.\n",
+    "- Fix variable to make it a square problem\n",
+    "- Run an initialization process.\n",
+    "- Solve the flowsheet.\n",
+    "\n",
+    "Following these steps, we've built a basic flowsheet using Pyomo. For more details, refer to the [documentation](../../flowsheets/hda_flowsheet_with_distillation.ipynb).\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 33,
+   "id": "f8e9968b",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "WARNING (W1001): Setting Var\n",
```

Comment:
Setting the initial values to a small number like `1e-5` instead of `0` may resolve these warnings.